controlplane/kubeadm/internal/cluster.go

```diff
@@ -206,7 +206,28 @@ func (m *ManagementCluster) RemoveEtcdMemberForMachine(ctx context.Context, clus
 		return err
 	}
 
-	return cluster.removeMemberForNode(ctx, machine.Status.NodeRef.Name)
+	controlPlaneNodes, err := cluster.getControlPlaneNodes(ctx)
+	if err != nil {
+		return err
+	}
+
+	nodeToRemove := machine.Status.NodeRef.Name
```

Comment:
I'm wondering if it would be good to add a check here that would throw an error if `controlPlaneNodes.Items < 2`, I can't think of any reason that we would remove the last etcd member for a cluster. @dlipovetsky thoughts?

Reply:
Because he validation webhook ensures the desired replicas is at least 1, I don't think it's necessary to add this check here. If this were an general etcd client, I would say that this function isn't the place for the check, but this client is internal, so I suppose it's fair game.

Reply:
agree with @dlipovetsky reason.

but rather than choose the "to be deleted" node as default (as you say here https://github.com/kubernetes-sigs/cluster-api/pull/2435#discussion_r384152901), how about iterate over the other nodes and pick one as default, and choose the "to be deleted" node only if `controlPlaneNodes.Items == 1`

Reply:
I'd add a comment to explain the reasoning here, so in the future we know we're relying on the validation functionality